napari/layers/base/base.py

```diff
@@ -52,6 +61,10 @@
 from napari.utils.transforms import Affine, CompositeAffine, TransformChain
 from napari.utils.translations import trans
 
+if TYPE_CHECKING:
+    import numpy.typing as npt
```

Comment:
Just as a final style issue, and this is totally up for discussion and not a requirement, but to consider: as an old-school Pythonista, I find that typing can get really convoluted very quickly. I think "English readability" is very important, and `npt.` is quite clunky. So what do people think about `from numpy.typing import ArrayLike` and then just using `ArrayLike`?  And you could add a comment next to it `# must be 1D`.

Reply:
I would say this would have to be discussed, perhaps open an issue. We use `npt` already like this in multiple locations, for example `test_transforms.py` and `_shapes_utils.py` so if we change the convention it needs to be changed in locations outside the intended scope of this PR.

Reply:
This is because `npt` follows conventions in the `numpy` docs, e.g. at https://numpy.org/doc/stable/reference/typing.html#number-precision.